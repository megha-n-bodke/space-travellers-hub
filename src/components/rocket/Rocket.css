.context {
  display: flex;
  flex-direction: row;
  justify-content: space-between;
  align-items: center;
  margin: 20px;
}

.img {
  width: 270px;
  height: 200px;
  object-fit: cover;
  margin: 10px;
}

.data {
  display: flex;
  flex-direction: column;
  justify-content: space-around;
  align-items: flex-start;
  margin: 10px;
  width: 100%;
  font-size: 20px;
}

.title {
  font-weight: bold;
  font-family: Poppins, sans-serif;
}

.btn-enabled {
  margin-top: 10px;
  height: 40px;
  width: 150px;
  background-color: rgb(0, 183, 255);
  border: none;
  border-radius: 5px;
  color: wheat;
  font-weight: bold;
}

.btn-disabled {
  margin-top: 10px;
  height: 40px;
  width: 150px;
  background-color: rgb(255, 13, 0);
  border: none;
  border-radius: 5px;
  color: wheat;
  font-weight: bold;
}

.sub-btn {
  display: flex;
  align-items: center;
  justify-content: center;
  margin: 10px 0;
  height: 20px;
  width: 100px;
  background-color: rgb(26, 133, 2);
  border: none;
  border-radius: 5px;
  color: wheat;
  font-weight: bold;
  font-size: 10px;
}
