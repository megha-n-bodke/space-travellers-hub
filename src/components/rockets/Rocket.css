.context{
    display: flex;
    flex-direction: row;
    justify-content: space-between;
    align-items: center;
    margin: 20px;
}

.img{
    width: 200px;
    height: 200px;
    object-fit: cover;
    margin: 10px;
}

.data{
    display: flex;
    flex-direction: column;
    justify-content: space-around;
    align-items: flex-start;
    margin: 10px;
}

.title{
    font-weight: bold;
    font-family: Poppins, sans-serif;
}

.dec{
    margin-top: -10px;
}

.btn {
    height: 40px;
    width: 150px;
    background-color: blue;
    border: none;
    color: wheat;
    font-weight: bold;
}

.btn:hover{
    background-color: rgb(1, 194, 4);
    cursor: pointer;
}