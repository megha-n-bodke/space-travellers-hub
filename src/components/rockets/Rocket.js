import React from 'react';
import './quote.css';

const Rocket = () => (
  <div className="container">
    <h1>Rocket</h1>
  </div>
);

export default Rocket;
