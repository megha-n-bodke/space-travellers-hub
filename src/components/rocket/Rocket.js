import React from 'react';
import './Rocket.css';
import PropTypes from 'prop-types';
import { useDispatch, useSelector } from 'react-redux';
import { cancelReservation, reserved } from '../../redux/profile/profile';

const Rocket = (props) => {
  const { name, image, description } = props;
  Rocket.propTypes = {
    name: PropTypes.string.isRequired,
    image: PropTypes.string.isRequired,
    description: PropTypes.string.isRequired,
  };

  const reserve = useSelector((state) => state.profile.reserved);
  const dispatch = useDispatch();
  const reservations = (e) => {
    const rocketTitle = e.target.parentElement.children[0].innerText;
    const rocket = rocketTitle;
    if (reserve.includes(rocket)) {
      return;
    }
    dispatch(reserved(rocket));
  };
  const cancel = (e) => {
    const rocketTitle = e.target.parentElement.children[0].innerText;
    dispatch(cancelReservation(rocketTitle));
  };
  return (
    <div className="context" data-testid="rocket">
      <img className="img" src={image} alt="user" />
      <div className="data">
        <h2 className="title">{name}</h2>
        { reserve.includes(name) && <div className="sub-btn">Reserved</div> }
        <p className="dec">
          {description}
        </p>
        {(reserve.includes(name)) ? (
          <button
            type="button"
            className="btn-disabled"
            onClick={cancel}
          >
            Cancel Reservations
          </button>
        ) : (
          <button
            type="button"
            className="btn-enabled"
            onClick={reservations}
          >
            Reserve
          </button>
        )}
      </div>
    </div>
  );
};

export default Rocket;
