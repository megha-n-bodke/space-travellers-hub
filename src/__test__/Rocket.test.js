import { render, screen } from '@testing-library/react';
import renderer from 'react-test-renderer';
import { Provider } from 'react-redux';
import '@testing-library/jest-dom';
import store from '../redux/configureStore';
import Rocket from '../components/rocket/Rocket';

it('Check any changes to the component', () => {
  const tree = renderer.create(
    <Provider store={store}>
      <Rocket />
    </Provider>,
  ).toJSON();
  expect(tree).toMatchSnapshot();
});

it('the component should render a div rocket', () => {
  render(
    <Provider store={store}>
      <Rocket />
    </Provider>,
  );

  const rocket = screen.getByTestId('rocket');

  expect(rocket).toBeInTheDocument();
});
