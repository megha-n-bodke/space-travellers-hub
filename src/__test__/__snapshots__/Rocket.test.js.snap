// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Check any changes to the component 1`] = `
<div
  className="context"
  data-testid="rocket"
>
  <img
    alt="user"
    className="img"
  />
  <div
    className="data"
  >
    <h2
      className="title"
    />
    <p
      className="dec"
    />
    <button
      className="btn-enabled"
      onClick={[Function]}
      type="button"
    >
      Reserve
    </button>
  </div>
</div>
`;
